python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -104,21 +106,31 @@ def update_usages(self):
         # And some computations are redundant. We should either remove redundant
         # computations or remove this method entirely and compute usages on demand.
         self._global_usage = ExecutionResources(0, 0, 0)
+        self._global_running_usage = ExecutionResources(0, 0, 0)
         self._op_usages.clear()
+        self._op_running_usages.clear()
         self._downstream_fraction.clear()
         self._downstream_object_store_memory.clear()
 
         # Iterate from last to first operator.
         num_ops_so_far = 0
         num_ops_total = len(self._topology)
         for op, state in reversed(self._topology.items()):
-            # Update `self._op_usages`.
+            # Update `self._op_usages` and `self._op_running_usages`.
             op_usage = op.current_processor_usage()
+            op_running_usage = op.current_running_processor_usage()
             assert not op_usage.object_store_memory
             op_usage.object_store_memory = self._estimate_object_store_memory(op, state)
+            op_running_usage.object_store_memory = self._estimate_object_store_memory(
```

Comment:
we should also add a similar `assert not op_running_usage.object_store_memory` check i think.

Reply:
Thank you, Scott, for the review. That makes sense. I'll update it.

Reply:
Hey @scottjlee, I've addressed your comment and would love your final review, thank you!